astropy/utils/data.py

```diff
@@ -1371,5 +1504,92 @@ def get_cached_urls():
         warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
         return False
 
-    with shelve.open(urlmapfn) as url2hash:
-        return list(url2hash.keys())
+    try:
```

Comment:
(Here and elsewhere) I'm not sure it is better to return an empty list, instead of just letting the error propagate...